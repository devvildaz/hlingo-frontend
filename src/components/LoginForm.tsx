import { loginResolver } from '@utils';
import {
  Button,
  Column,
  FormControl,
  Input,
  KeyboardAvoidingView,
  useToast,
  WarningOutlineIcon,
} from 'native-base';
import { useState } from 'react';
import { useForm, Controller } from 'react-hook-form';

interface FormData {
  email: string;
  password: string;
}

interface Props {
  login: (data: FormData) => Promise<string>;
}

const LoginForm = ({ login }: Props) => {
  const { handleSubmit, control, formState: { errors }} = useForm<FormData>({
    defaultValues: {
      email: '',
      password: '',
    },
    resolver: loginResolver,
    mode: 'onSubmit',
  }); // prettier-ignore
  const [isLoading, setIsLoading] = useState(false);

  const toast = useToast();

  const onSubmit = async (data: FormData) => {
    setIsLoading(true);
    try {
      await login(data);
    } catch (error) {
      console.log(error);
      toast.show({
        title: 'Algo salió mal',
        placement: 'bottom',
        bgColor: 'red.500',
        color: 'white',
      });
    } finally {
      setIsLoading(false);
    }
  };

  return (
    <KeyboardAvoidingView py={10} px={8} width="full" maxW="sm" mx="auto">
      <Column space={3}>
        {/* Email */}
        <FormControl isInvalid={Boolean(errors.email)}>
          <FormControl.Label>Correo electrónico</FormControl.Label>
          <Controller
            name="email"
            control={control}
            render={({ field: { onChange, onBlur, value } }) => (
              <Input
                onBlur={onBlur}
                onChangeText={value => onChange(value)}
                value={value}
                type="text"
                keyboardType="email-address"
                placeholder="Ingrese su correo..."
                accessibilityLabel="Correo electrónico"
              />
            )}
          />
          {errors.email != null && (
            <FormControl.ErrorMessage
              leftIcon={<WarningOutlineIcon />}
              accessibilityRole="alert"
            >
              {errors.email.message}
            </FormControl.ErrorMessage>
          )}
        </FormControl>

        {/* Password */}
        <FormControl isInvalid={Boolean(errors.password)}>
          <FormControl.Label>Contraseña</FormControl.Label>
          <Controller
            name="password"
            control={control}
            render={({ field: { onChange, onBlur, value } }) => (
              <Input
                onBlur={onBlur}
                onChangeText={value => onChange(value)}
                value={value}
                type="password"
                placeholder="Ingrese su contraseña..."
                accessibilityLabel="Contraseña"
              />
            )}
          />
          {errors.password != null && (
            <FormControl.ErrorMessage
              leftIcon={<WarningOutlineIcon />}
              accessibilityRole="alert"
            >
              {errors.password.message}
            </FormControl.ErrorMessage>
          )}
        </FormControl>

        <Button
          // eslint-disable-next-line @typescript-eslint/no-misused-promises
          onPress={handleSubmit(onSubmit)}
          mt={2}
          isLoading={isLoading}
          accessibilityRole="button"
        >
          Iniciar Sesión
        </Button>
      </Column>
    </KeyboardAvoidingView>
  );
};

export default LoginForm;
