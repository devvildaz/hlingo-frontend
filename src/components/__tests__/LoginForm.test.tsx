import React from 'react';
import renderer from 'react-test-renderer';

import LoginForm from '@components/LoginForm';
import Wrapper from '../ProviderWrapper';

describe('<LoginForm />', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  it('should match the snapshot', () => {
    const component = renderer.create(
      <Wrapper>
        <LoginForm />
      </Wrapper>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
