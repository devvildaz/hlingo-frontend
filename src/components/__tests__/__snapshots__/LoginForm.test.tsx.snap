// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    onLayout={[Function]}
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": 384,
        "paddingBottom": 40,
        "paddingLeft": 32,
        "paddingRight": 32,
        "paddingTop": 40,
        "width": "100%",
      }
    }
  >
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <View
        dataSet={Object {}}
        isInvalid={false}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          dataSet={Object {}}
          feedbackId="field-react-aria-1-feedback"
          hasFeedbackText={false}
          hasHelpText={false}
          helpTextId="field-react-aria-1-helptext"
          isDisabled={false}
          isInvalid={false}
          isReadOnly={false}
          isRequired={false}
          labelId="field-react-aria-1-label"
          nativeID="field-react-aria-1-label"
          setHasFeedbackText={[Function]}
          setHasHelpText={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginBottom": 4,
              "marginTop": 4,
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#737373",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "500",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Correo electrónico
          </Text>
        </View>
        <View
          dataSet={Object {}}
          isFocused={false}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#d4d4d4",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "overflow": "hidden",
            }
          }
        >
          <TextInput
            accessible={true}
            allowFontScaling={true}
            dataSet={Object {}}
            disabled={false}
            editable={true}
            isRequired={false}
            nativeID="field-react-aria-1"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Ingrese su correo..."
            placeholderTextColor="#a3a3a3"
            readOnly={false}
            rejectResponderTermination={true}
            required={false}
            secureTextEntry={false}
            selectionColor="#1f2937"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#171717",
                "flex": 1,
                "fontFamily": undefined,
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "400",
                "paddingBottom": 8,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 12,
          }
        }
      />
      <View
        dataSet={Object {}}
        isInvalid={false}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          dataSet={Object {}}
          feedbackId="field-react-aria-2-feedback"
          hasFeedbackText={false}
          hasHelpText={false}
          helpTextId="field-react-aria-2-helptext"
          isDisabled={false}
          isInvalid={false}
          isReadOnly={false}
          isRequired={false}
          labelId="field-react-aria-2-label"
          nativeID="field-react-aria-2-label"
          setHasFeedbackText={[Function]}
          setHasHelpText={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginBottom": 4,
              "marginTop": 4,
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#737373",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "500",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Contraseña
          </Text>
        </View>
        <View
          dataSet={Object {}}
          isFocused={false}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#d4d4d4",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "overflow": "hidden",
            }
          }
        >
          <TextInput
            accessible={true}
            allowFontScaling={true}
            dataSet={Object {}}
            disabled={false}
            editable={true}
            isRequired={false}
            nativeID="field-react-aria-2"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="*************"
            placeholderTextColor="#a3a3a3"
            readOnly={false}
            rejectResponderTermination={true}
            required={false}
            secureTextEntry={true}
            selectionColor="#1f2937"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#171717",
                "flex": 1,
                "fontFamily": undefined,
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "400",
                "paddingBottom": 8,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 8,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 12,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={Object {}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0891b2",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 8,
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
          test={true}
        >
          <View
            dataSet={Object {}}
            style={Object {}}
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#fafafa",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                }
              }
            >
              Iniciar Sesión
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
